File: text-recognition/google-cloud-vision/GoogleCloudVisionTextRecognition.java
// "input" is a string holding the path to the image.

Path path = Paths.get(input);
byte[] bytes = Files.readAllBytes(path);
ByteString content = ByteString.copyFrom(bytes);

Image image = Image.newBuilder()
    .setContent(content)
    .build();
Feature feature = Feature.newBuilder()
    .setType(Feature.Type.TEXT_DETECTION)
    .build();
AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
    .addFeatures(feature)
    .setImage(image)
    .build();

List<AnnotateImageRequest> requests = new ArrayList<>();
requests.add(request);

try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
    BatchAnnotateImagesResponse batchResponse = client.batchAnnotateImages(requests);
    AnnotateImageResponse response = batchResponse.getResponsesList().get(0);

    List<EntityAnnotation> textAnnotationsList = response.getTextAnnotationsList();
    TextAnnotation fullTextAnnotation = response.getFullTextAnnotation();
}

File: text-recognition/google-cloud-vision/GoogleCloudVisionTextRecognition.kt
// "input" is a File object pointing to an image file.

val bytes = ByteString.copyFrom(input.readBytes())

val image = Image.newBuilder()
    .setContent(bytes)
    .build()
val feature = Feature.newBuilder()
    .setType(Feature.Type.TEXT_DETECTION)
    .build()
val request = AnnotateImageRequest.newBuilder()
    .addFeatures(feature)
    .setImage(image)
    .build()

val requests = listOf(request)

val client = ImageAnnotatorClient.create()

val response = client.batchAnnotateImages(requests)
val responses = response.responsesList.single()

val textAnnotationsList = responses.textAnnotationsList
val fullTextAnnotation = responses.fullTextAnnotation
